internal/blockchain/fabric/fabric.go

```diff
@@ -691,13 +635,15 @@ func (f *Fabric) SubmitBatchPin(ctx context.Context, nsOpID string, signingKey s
 		"contexts":   hashes,
 	}
 	input, _ := jsonEncodeInput(pinInput)
-	f.fireflyContract.mux.Lock()
-	chaincode := f.fireflyContract.chaincode
-	f.fireflyContract.mux.Unlock()
-	return f.invokeContractMethod(ctx, f.defaultChannel, chaincode, batchPinMethodName, signingKey, nsOpID, batchPinPrefixItems, input, nil)
+	return f.invokeContractMethod(ctx, f.defaultChannel, fabricOnChainLocation.Chaincode, batchPinMethodName, signingKey, nsOpID, batchPinPrefixItems, input, nil)
```

Comment:
I don't think we should assume "defaultChannel" here - I think we should use the channel and chaincode from the passed-in location.